### PR TITLE
WIP: RFC-10 - Connections

### PR DESCRIPTION
**Pull request by piogor**

### CODE REVIEW

File: 10/RequestReply.md [hunk 1]
```diff
@@ -103,14 +103,40 @@ NOTE All GROUPED REQUESTS should be logged once with the same information as the
 
 ### Connection management
 
-* The Reqest-Reply protocol SHALL manage client connection.
+The Reqest-Reply protocol SHALL manage client connection to devcie server.
```
Comment (piogor) [inline 10/RequestReply.md]:
```suggestion
The Request-Reply protocol SHALL manage client connections to Device Servers.

File: 10/RequestReply.md [hunk 2]
```diff
@@ -103,14 +103,40 @@ NOTE All GROUPED REQUESTS should be logged once with the same information as the
...
+  network endpoints like operating system inter-process communication mechanism.
+* A Client MAY get information about Device Server's endpoints:
+  * From the Tango Database service, if the service is available,
+  * From command-line parameters,
+  * Other way,
+* The endpoint of Database Devcie Server SHOULD be provided to the Client via the `TANGO_HOST` environment variable defined as follows:
+... abnf
+ tango_host = network_address ":" tcp_port 
+...
+
```
Comment (piogor) [inline 10/RequestReply.md]:
```suggestion
* To send any REQUESTS to a Device, a Client MUST connect to the Device Server manging the Device.  
* A Device Server SHALL wait for client connections on one or multiple network endpoints.
* The network endpoint SHALL define \<network_address\> and \<tcp_port\>, where:

... abnf
 network_address = ip_address / host_name
 tcp_port = num-val
...

* A Client MAY connect any Device Server on one of Device Server's endpoints.
* If a Client is running on the same host as a Device Server, the client MAY use other than
  network endpoints like operating system inter-process communication mechanism.
* A Client MAY get information about Device Server's endpoints:
  * From the Tango Database service, if the service is available,
  * From command-line parameters,
  * In another suitable way,
* The endpoint of Database Device Server SHOULD be provided to the Client via the `TANGO_HOST` environment variable defined as follows:
... abnf
 tango_host = network_address ":" tcp_port 
...

```

File: 10/RequestReply.md [hunk 3]
```diff
@@ -103,15 +103,40 @@ NOTE All GROUPED REQUESTS should be logged once with the same information as the
...
+* A Device Server SHALL wait for client connections on one or multiple network endpoints.
+* The network endpoint SHALL define \<network_address\> and \<tcp_port\>, where:
+
+... abnf
+ network_address = ip_address / host_name
+ tcp_port = num-val
+...
+
+* A Client MAY connect any Device Server on one of Device Server's endpoints.
+* If a Client is running on the same host as a Device Server, the client MAY use other than
```
Comment (hardion) [inline 10/RequestReply.md:120]: This is specific to the CORBA protocol so I would say the way to connect is delegated to the underline protocol. It means that this RFC should say explicitly what is an underline protocol.

File: 10/RequestReply.md [hunk 4]
```diff
@@ -103,15 +103,40 @@ NOTE All GROUPED REQUESTS should be logged once with the same information as the
...
+  network endpoints like operating system inter-process communication mechanism.
+* A Client MAY get information about Device Server's endpoints:
+  * From the Tango Database service, if the service is available,
```
Comment (hardion) [inline 10/RequestReply.md:123]: I would rather say that the Tango Database specify the way to get the connection information. Other way have to be specify by the Tango Database RFC.

File: 10/RequestReply.md [hunk 5]
```diff
@@ -103,15 +103,40 @@ NOTE All GROUPED REQUESTS should be logged once with the same information as the
...
+  * From command-line parameters,
+  * In another suitable way,
```
Comment (hardion) [inline 10/RequestReply.md:125]: This is not specific enough and can lead to confusion.

File: 10/RequestReply.md [hunk 6]
```diff
@@ -103,15 +103,40 @@ NOTE All GROUPED REQUESTS should be logged once with the same information as the
...
+* The network endpoint SHALL define \<network_address\> and \<tcp_port\>, where:
+
+... abnf
+ network_address = ip_address / host_name
+ tcp_port = num-val
+...
+
+* A Client MAY connect any Device Server on one of Device Server's endpoints.
+* If a Client is running on the same host as a Device Server, the client MAY use other than
+  network endpoints like operating system inter-process communication mechanism.
```
Comment (piogor) [inline 10/RequestReply.md:121]:
```suggestion
```

File: 10/RequestReply.md (hunk 3 above)
Comment (piogor) [reply 10/RequestReply.md:120]: So, I propose to remove this point.

File: 10/RequestReply.md (hunk 4 above)
Comment (piogor) [reply 10/RequestReply.md:123]: I look on [the RFC-6](https://github.com/tango-controls/rfc/blob/draft/6/Database.md#exported-devices) and I thinkit is worth to add that the Device Server SHALL call DbExportDevice to publish the endpoint and DbUnExportDevice to unpublish.

Comment (hardion) [reply 10/RequestReply.md:123]: Good point! In this case we should emphasis the role of the Device Server to use the  Request-Reply to register the new connection.

**Pull request closed without merging**